api/client.go

```diff
@@ -1249,6 +1249,83 @@ START:
 	return result, nil
 }
 
+// requestWithContext avoids the use of the go-retryable library found in RawRequestWithContext and is
+// useful when making requests where the req or resp body is likely larger than available memory
+func (c *Client) requestWithContext(ctx context.Context, r *Request) (*Response, error) {
+	req, err := http.NewRequestWithContext(ctx, r.Method, r.URL.RequestURI(), r.Body)
```

Comment:
Curious why you don't use NewRequest?

Reply:
To be clear, I meant Client.NewRequest, not http.NewRequest.

Reply:
The client function `ToHTTP` has been deprecated (likely because it's encouraged to use `toRetryableHTTP()`) and in this case, I think it's more clear/succinct to see that we are using an `*http.Request` in this function.